### PR TITLE
Fix and extend caching documentation

### PR DESCRIPTION
**Pull request by cb372**
Fixed a few incorrect statements. Added more information about the interaction between TTL, `max-age` and `ETag`.

### CODE REVIEW

File: source/caching.md.erb [hunk 1]
```diff
@@ -67,13 +67,131 @@ If this header is set on an API response, Octoparts will guarantee not to insert
 
 If `max-age` is set, Octoparts will cache the response with the given TTL (in seconds). However, if a TTL is also set in the admin screen for that endpoint, the shorter of the two TTLs will be used.
 
+Note that if the response can be revalidated (i.e. it has an `ETag` or `Last-Modified` header), then the `max-age` value will NOT be used to decide the cache TTL. The TTL value configured in the admin UI (which may be eternal) will be used unconditionally.
```
Comment (rinoguchi) [inline source/caching.md.erb]: I think "The TTL value set in the admin screen" is better because there is the same words in previous line.

Comment (cb372) [inline source/caching.md.erb]: Good point. Fixed.

**Comment by lloydmeta** [comment]
[LGTM](http://www.lgtm.in/i/eLQVBNNSp)

[:+1:](http://www.lgtm.in/u/eLQVBNNSp) [:-1:](http://www.lgtm.in/r/eLQVBNNSp)

File: source/caching.md.erb [hunk 2]
```diff
@@ -67,13 +67,131 @@ If this header is set on an API response, Octoparts will guarantee not to insert
...
+If one or more of these headers are set, Octoparts will store their values together with the response itself in the cache.
+
+The next time Octoparts looks up the response in the cache, it will contact the backend API again in order to validate that the cached response is not stale. It will send `If-None-Match` and/or `If-Modified-Since` headers. If the backend responds with `304 Not Modified`, Octoparts will return the cached response to the client. Otherwise it will cache the new response and return it to the client.
+
+Note that if the the response also included a `max-age` directive, revalidation will not occur until after this period has elapsed. For example, if the backend response had both an `ETag` and a `max-age` of 60 seconds, Octoparts will treat the response as fresh for 60 seconds. During the first 60 seconds it will not bother to revalidate the response, but after 60 seconds have elapsed it will start revalidating using the `ETag` value.
+
+#### Revalidation and TTL reset
```
Comment (rinoguchi) [inline source/caching.md.erb:80]: You'd better say that in this case the TTL can be set in the admin screen only.

Comment (mauhiz) [inline source/caching.md.erb:80]: @rinoguchi sorry for merging before you finished reviewing. @cb372 will fix on develop :)

Comment (rinoguchi) [inline source/caching.md.erb:80]: No problem. Thank you!

**Pull request merged**